crates/typst-cli/src/compile.rs

```diff
@@ -84,8 +89,13 @@ pub fn compile_once(
                 Status::Error.print(command).unwrap();
             }
 
-            print_diagnostics(world, *errors, warnings, command.diagnostic_format)
-                .map_err(|_| "failed to print diagnostics")?;
+            print_diagnostics(
+                world,
+                *errors,
+                warnings.to_vec(),
```

Comment:
The `to_vec()`s here introduce an ugly copy, but I wanted to avoid adding another dependency to this crate